crates/hyperqueue/src/dashboard/ui/screens/job/job_tasks_chart.rs

```diff
@@ -0,0 +1,155 @@
+use crate::dashboard::data::job_timeline::DashboardTaskState;
+use crate::dashboard::data::DashboardData;
+use crate::dashboard::ui::terminal::DashboardFrame;
+use crate::dashboard::ui::widgets::chart::{ChartPlotter, DashboardChart, PlotStyle};
+use crate::JobId;
+use std::time::{SystemTime, UNIX_EPOCH};
+use tako::common::Map;
+use tui::layout::Rect;
+use tui::style::Color;
+use tui::symbols::Marker;
+
+const RUNNING_TAG: &str = "Running";
+const FINISHED_TAG: &str = "Finished";
+const FAILED_TAG: &str = "Failed";
+
+#[derive(Default)]
+pub struct JobTaskChart {
+    /// The Job for which we want to see the tasks for.
+    job_id: Option<JobId>,
+    /// The chart.
+    chart: DashboardChart,
+}
+
+impl JobTaskChart {
+    /// Sets a new job id for the chart to plot data for.
+    pub fn set_job_id(&mut self, job_id: JobId) {
+        match self.job_id {
+            Some(chart_job_id) if chart_job_id == job_id => (),
+            _ => {
+                self.job_id = Some(job_id);
+                self.chart.clear_chart();
```

Comment:
Can we just replace the whole chart? Unless it's really needed, it's nice to avoid such heavily mutable APIs (clear + fill).